vmm/src/migration.rs

```diff
@@ -49,10 +49,12 @@ pub fn recv_vm_config(source_url: &str) -> std::result::Result<VmConfig, Migrata
     vm_config_path.push(SNAPSHOT_CONFIG_FILE);
 
     // Try opening the snapshot file
-    let vm_config_file =
+    let mut vm_config_file =
         File::open(vm_config_path).map_err(|e| MigratableError::MigrateSend(e.into()))?;
-    let vm_config_reader = BufReader::new(vm_config_file);
-    serde_json::from_reader(vm_config_reader).map_err(|e| MigratableError::MigrateReceive(e.into()))
+    let mut bytes = Vec::new();
+    let _ = vm_config_file.read_to_end(&mut bytes);
```

Comment:
We should handle the error explicitly.

Reply:
Yes, well caught @likebreath we shouldn't regress on the error handling here - particularly as the JSON data could easily get corrupted.

Reply:
Thx, patch will be updated.